src/components/TimetablePage/DefaultPage/index.tsx

```diff
@@ -128,19 +128,27 @@ function CurrentSemesterLectureList({
   const { data: myLecturesFromServer } = useTimetableInfoList(selectedSemester, token);
   const { mutate: mutateAddWithServer } = useAddTimetableLecture(token);
   const isLoaded = status === 'success' && (myLecturesFromLocalStorageValue !== null || myLecturesFromServer !== undefined);
+
   return (
     isLoaded ? (
       <LectureTable
         height={459}
         list={
           (lectureList as unknown as Array<LectureInfo>)
-            .filter(
-              (lecture) => (
-                lecture.name.includes(filter.search)
-                // 백엔드 수정하면 제거
-                // || (filter.dept !== '전체' && lecture.department === filter.dept)
-              ),
-            )
+            .filter((lecture) => {
+              const searchFilter = filter.search;
+              const deptFilter = filter.dept;
+              if (searchFilter && deptFilter === '전체') {
```

Comment:
truthly한 값으로 사용하자니 모호해 보리는데, searchFilter의 타입을 기반으로 명확하게 작성하는것도 좋을 것 같아요~

Reply:
고민하다가 지저분해보여서 그냥 truthly 하게 표현했는데, 만약 바꾼다면 이렇게 되려나요 👀 
```suggestion
              if (searchFilter !== '' && deptFilter === '전체') {
```

Reply:
네넹 맞아여 `0, null, undefined, []` 등을 배제할 수 있겠죠

Reply:
근데 큰 차이가 있는건 아닌 마이너한 리뷰라.. 그냥 넘겨도 전혀 지장없습니다!

Reply:
다시 보니 훨씬 명시적이라서 좋은 것 같네요. deptfilter 조건과도 통일성이 있기에 변경했습니다. 206a0ebb57f868616b6d72119e24d3e3ad49b860